src/main/java/uk/gov/hmcts/reform/sscs/helper/mapping/HearingsPartiesMapping.java

```diff
@@ -451,19 +419,42 @@ public static List<UnavailabilityDayOfWeek> getDwpUnavailabilityDayOfWeek() {
         return getPartyUnavailabilityDayOfWeek();
     }
 
-    public static List<UnavailabilityRange> getPartyUnavailabilityRange(HearingOptions hearingOptions) {
+    public static List<UnavailabilityRange> getPartyUnavailabilityRange(HearingOptions hearingOptions) throws ListingException {
         if (isNull(hearingOptions) || isNull(hearingOptions.getExcludeDates())) {
             return Collections.emptyList();
         }
 
-        return hearingOptions.getExcludeDates().stream()
-                .map(ExcludeDate::getValue)
-                .map(dateRange -> UnavailabilityRange.builder()
-                        .unavailableFromDate(LocalDate.parse(dateRange.getStart()))
-                        .unavailableToDate(LocalDate.parse(dateRange.getEnd()))
-                        .unavailabilityType(ALL_DAY.getLabel())
-                        .build())
-                .collect(Collectors.toList());
+        List<UnavailabilityRange> unavailabilityRanges = new LinkedList<>();
+
+        for (ExcludeDate excludeDate : hearingOptions.getExcludeDates()) {
+            UnavailabilityRange unavailabilityRange = getUnavailabilityRange(excludeDate.getValue());
+
+            if (nonNull(unavailabilityRange)) {
+                unavailabilityRanges.add(unavailabilityRange);
+            }
+        }
+
+        return unavailabilityRanges;
+    }
+
+    private static UnavailabilityRange getUnavailabilityRange(DateRange dateRange) throws ListingException {
+        LocalDate startDate = dateRange.getStartDate();
+
+        if (isNull(startDate)) {
+            return null;
+        }
+
+        LocalDate endDate = dateRange.getEndDate();
+
+        if (endDate.isBefore(startDate)) {
```

Comment:
Will this return null or throw exception if end date is null or not valid date format?

Reply:
Cool, so we're checking against this outcome via common changes. Good stuff.